dbt/adapters/base/relation.py

```diff
@@ -225,6 +239,31 @@ def render_limited(self) -> str:
         else:
             return f"(select * from {rendered} limit {self.limit}){self._render_limited_alias()}"
 
+    def render_event_time_filtered(self, rendered: Optional[str] = None) -> str:
+        rendered = rendered or self.render()
+        if self.event_time_filter is None:
+            return rendered
+
+        filter = self._render_event_time_filtered(self.event_time_filter)
+        if not filter:
+            return rendered
+
+        return f"(select * from {rendered} where {filter}){self._render_subquery_alias(namespace='et_filter')}"
+
+    def _render_event_time_filtered(self, event_time_filter: EventTimeFilter) -> str:
+        """
+        Returns "" if start and end are both None
+        """
+        filter = ""
+        if event_time_filter.start and event_time_filter.end:
+            filter = f"{event_time_filter.field_name} >= '{event_time_filter.start}' and {event_time_filter.field_name} < '{event_time_filter.end}'"
+        elif event_time_filter.start:
+            filter = f"{event_time_filter.field_name} >= '{event_time_filter.start}'"
+        elif event_time_filter.end:
+            filter = f"{event_time_filter.field_name} < '{event_time_filter.end}'"
```

Comment:
Are we sure we want to mix inclusive/exclusive comparisons?

Reply:
I think so. The way we're going to be chunking up microbatch runs will be by partitions. For example, we'll be building daily partition `2020-01-01 ( 2020-01-01 > event_time <= 2020-01-02)`, followed by `2020-01-02 ( 2020-01-02 > event_time <= 2020-01-03)`, so these inclusive/exclusive semantics work very intuitively with that. 

https://github.com/dbt-labs/dbt-core/pull/10677 (check out the preliminary screenshots there!)